index.go

```diff
@@ -119,6 +119,12 @@ func (b *Batch) Reset() {
 	b.internal.Reset()
 }
 
+func (b *Batch) Merge(o *Batch) {
+	b.internal.Merge(o.internal)
+	b.lastDocSize = o.LastDocSize()
+	b.totalSize = uint64(b.internal.TotalDocSize())
```

Comment:
one edge case is if the incoming 'o' batch is empty?  In that case, taking o's lastDocSize and totalSize might be wrong?

Reply:
lastDocSize might be wrong, but totalSize is computed from updated b. So may not change.

Reply:
> but totalSize is computed from updated b.

ah, didn't catch that -- thx!